tests/kernel/dsp_sharing/generic/src/complex_calculation.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2022 Synopsys
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @file
+ * @brief complex number multiplication portion of DSP sharing test
+ *
+ * @ingroup kernel_dspsharing_tests
+ *
+ * This module is used for the DSP sharing test, and supplements the basic
+ * load/store test by incorporating two additional threads that utilize the
+ * DSP unit.
+ *
+ * Testing utilizes a pair of tasks that independently compute complex vector
+ * dot product. The lower priority task is regularly preempted by the higher
+ * priority task, thereby testing whether DSP context information is properly
+ * preserved.
+ *
+ * A reference value of computed result is computed once at the start of the
+ * test. All subsequent computations must produce the same value, otherwise
+ * an error has occurred.
+ */
+
+#include <ztest.h>
+#include "fxarc.h"
```

Comment:
ARC specific, add some #ifdef? Also for the later code that uses the ARC specific features.
Alternatively, make this an arch specific test.

Reply:
fixed